policy-controller/k8s/index/src/index.rs

```diff
@@ -1021,6 +1032,7 @@ impl Pod {
     fn default_inbound_server(
         port: NonZeroU16,
         settings: &pod::Settings,
+        probes: &PortMap<HashSet<String>>,
```

Comment:
just like with http_probe_routes, we're only using one entry from this PortMap.  so I think we can similarly change this to `probe_paths: impl Iterator<Item = &'p str>` and pull the PortMap lookup up into the calling function.

Reply:
The reason I didn't make the change here as well is because `port` is already an argument, so passing a reference to the HashMap felt fine. Where a similar change was made, `probes` and `port` were new argument and `port` was only included as a a key for `probes` — so it seemed best to change that in the callsite.

In this case we're iterating through `probes` at some point so it seemed fine to do here.

Reply:
Even though we use the `port` for other things as well, I find it harder to understand what a function is doing when it accepts more data than it needs.  In this case we're giving it the entire PortMap when all it cares about are the paths for a specific port.

Reply:
That's a good point. I changed it.